test/core/end2end/tests/compressed_payload.c

```diff
@@ -193,6 +193,7 @@ static void request_for_disabled_algorithm(
   GPR_ASSERT(GRPC_CALL_OK == error);
 
   CQ_EXPECT_COMPLETION(cqv, tag(101), true);
+  CQ_EXPECT_COMPLETION(cqv, tag(1), true);
```

Comment:
I don't understand this change.  How can the client batch containing recv ops be complete at this point, since the server hasn't yet sent anything?

Reply:
I think because the server disabled the particular compression algorithm (note the name of this function), server will cancel the call resulting this batch to complete.

Reply:
If you look at https://github.com/grpc/grpc/pull/12750/files/39f0895c021980d7b644e745fe743e697f5fad00#diff-062e65d2ed8816ecdef192091b7cf7fdR213, you'll notice that we don't expect the server to perform those ops. So, on establishing the channel, the client probably gets notified that compression is not supported and hence gets canceled.